tests/unit/lms/services/canvas_api/client_test.py

```diff
@@ -383,6 +385,45 @@ def test_list_duplicate_files(self, canvas_api_client, http_session):
 
         assert response == [files[0]]
 
+    def test_list_files_emits_event(self, canvas_api_client, http_session, registry):
+        # pylint: disable=protected-access
+        canvas_api_client._request.registry = registry
+        files = [
+            {
+                "display_name": "display_name_1",
+                "id": 1,
+                "updated_at": "updated_at_1",
+                "size": 12345,
+            },
+            {
+                "display_name": "display_name_1",
+                "id": 2,
+                "updated_at": "updated_at_1",
+                "size": 12345,
+            },
+        ]
```

Comment:
```suggestion
        files = [
            {
                "id": i,
                "display_name": "display_name_{i}",
                "updated_at": "updated_at_{i}",
                "size": 1000+i,
            } for i in range(2)
        ]
```

It's probably a good idea to vary the things we are asserting against here.